tree_unique_args/src/type_analysis.rs

```diff
@@ -1,3 +1,6 @@
+#[cfg(test)]
+const SCHED: &str = "(run-schedule (repeat 5 (saturate type-analysis) (saturate always-run)))";
```

Comment:
Do you need this schedule?
I think we should add all your typechecking rules to the normal schedule, and `run_test` will just work